commons/src/types/Availability.ts

```diff
@@ -97,6 +101,19 @@ export interface AvailabilityQuery extends CommonQuery {
   forceReload?: boolean;
 }
 
+export interface ConsecutiveAvailabilityQuery extends CommonQuery {
```

Comment:
Not really related to this PR, but I'm starting to dislike this pattern. It requires a lot of boilerplating and doesn't scale well compared to just passing parameters for our API requests.

Reply:
Yep, definitely with you on this one after tonight. Type convolution get really problematic with this PR in particular, because we're doing a lot of appending of event definition info into timeslots. Will try to make some improvements.